relayerfactory.go

```diff
@@ -13,11 +12,15 @@ import (
 	"go.uber.org/zap"
 )
 
+type TestName interface {
+	Name() string
+}
+
 // RelayerFactory describes how to start a Relayer.
 type RelayerFactory interface {
 	// Build returns a Relayer associated with the given arguments.
 	Build(
-		t *testing.T,
+		t TestName,
```

Comment:
@DavidNix 
See any issues here? 
Reece notes why this is necessary in the description. Would be good to run it by you too.

Reply:
This is fine. There is a new interface on line 15 `TestName` of which `*testing.T` implements. 

I'm betting this abstraction allows us to not care about creating an actual `*testing.T` when, in the context of this feature, you don't need because you are not making test assertions. Rather, you're trying to spin up a "mini-network" on your machine.